### ISSUE TITLE
Execution cancelled when setting an array element to an object

### ISSUE THREAD

**Issue by cristianbuse** [WaynePhillipsEA/twinbasic#274]
**To Reproduce**
```
Module Test
    Public Sub Main()
        Dim obj As Object
        Dim arr() As Variant
        ReDim arr(0 To 1)
        
        Set arr(1) = obj    'Execution is cancelled here
        Debug.Print "Done"
    End Sub
End Module
```
**Desktop (please complete the following information):**
 - OS: Windows 10 Enterprise V1909
 - twinBASIC compiler v0.10.3423

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.3429, thanks for reporting!

**Comment by cristianbuse** [comment]
If I change the variable type from ```Object``` to ```ErrObject``` the execution is still cancelled. What is absolutely weird is that in this case I cannot even step through the code line by line.

```VBA
Module Test
    Public Sub Main()
        Dim obj As ErrObject
        Dim arr() As Variant
        ReDim arr(0 To 1)
        
         Set arr(1) = obj
        Debug.Print "Done"
    End Sub
End Module
```

**Comment by cristianbuse** [comment]
@WaynePhillipsEA Can you please re-open this issue so that it does not go off the radar? Thanks!

**Comment by WaynePhillipsEA** [comment]
Thanks @cristianbuse I'll look at that one.  It's probably due to ErrObject and Collection still using the very early implementations of IDispatch. They need to move to the stable versions as used by normal VBA classes, but I'll double check that.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.3613, thanks for reporting.  There will likely still be some edge cases involving ErrObject, as it is special-cased internally for optimization purposes, and we haven't got enough test coverage for it yet.

**Comment by cristianbuse** [comment]
Thanks for fixing that. Indeed you are right, there is one edge case I immediately found. There is no problem if the ```obj``` variable is declared within the function scope but being module level causes issues:  
```VBA
Module Test
    Private obj As ErrObject
    
    Public Sub Main()
        Dim coll As Collection = New Collection
        
        coll.Add obj 'Execution is cancelled here
        
        Debug.Print "Done"
    End Sub
End Module
```

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.3664, but I know there are more ;)

**Comment by cristianbuse** [comment]
Yes, here are a few more:

1. If you add a ```New``` keyword to the above code by changing the ```Private obj As ErrObject``` to ```Private obj As New ErrObject``` we get a crash of compiler and even got VSCode crashed. It seems that in VBA this is valid code and it only throws a runtime error saying "Class not registered".

2. Code in ```Test``` method never gets called:  
```VBA
Module Test
    Public Sub Main()
        Debug.Print Test(Err).Number
    End Sub
    
    Private Function Test(ByRef e As ErrObject) As ErrObject
        Err.Raise 5 'Never gets raised
        Set Test = e
    End Function
End Module
```
doesn't matter if ```e``` is passed ```ByVal``` or ```ByRef```

3. This one throws a compilation error:
```VBA
Module Test
    Public Sub Main()
        Debug.Print Test(Err).Number
    End Sub
    
    Private Function Test(ByVal e As Object) As ErrObject
        Set Test = e
    End Function
End Module
```

**Comment by WaynePhillipsEA** [comment]
Thanks @cristianbuse, those ones should now be fixed in v0.10.3685.
